test/components/lazy-load.spec.js

```diff
@@ -20,86 +20,145 @@ describe('LazyLoad spec', () => {
+
+        instance.onScroll()
+
+        expect(stub.calledOnce).to.be.true
+      }))
+    })
+
+    context('when element doesn\'t have length', () => {
+      it('should remove scroll event from window', sinon.test(function () {
+        const stub = this.stub(instance, 'onScrollHandler')
+
+        instance.element = ''
+        instance.onScroll()
+
+        triggerEvent(window, 'scroll')
+
+        expect(stub.calledOnce).to.be.not.true
+      }))
+    })
   })
 
-  describe('checkPlaceholder', () => {
-    it('should call renderImage and if placeholder is visible', sinon.test(function () {
-      this.stub(instance, 'isPlaceholderVisible').returns(true)
-      this.spy(instance, 'renderImage')
+  describe('@checkVisiblePlaceholders', () => {
+    it('should have a windowHeight property', () => {
+      instance.checkVisiblePlaceholders()
 
-      instance.checkPlaceholder(instance.$element[0])
+      expect(instance).to.have.property('windowHeight')
+    })
 
-      expect(instance.renderImage.calledOnce).to.be.true
-    }))
+    it('should have a windowWidth property', () => {
+      instance.checkVisiblePlaceholders()
+
+      expect(instance).to.have.property('windowWidth')
+    })
 
-    it('should not call renderImage and if placeholder is not visible', sinon.test(function () {
-      this.stub(instance, 'isPlaceholderVisible').returns(false)
-      this.spy(instance, 'renderImage')
+    it('should call @checkPlaceholder', sinon.test(function () {
+      const spy = this.spy(instance, 'isPlaceholderVisible')
 
-      instance.checkPlaceholder()
+      instance.checkVisiblePlaceholders()
 
-      expect(instance.renderImage.notCalled).to.be.true
+      expect(spy.calledWith(instance.element[0])).to.be.true
     }))
   })
 
-  describe('renderImage', () => {
-    it('should call parseBreakpoints if the image is a srcset image', sinon.test(function () {
-      this.stub(instance, 'isPlaceholderVisible').returns(true)
-      this.spy(instance, 'parseBreakpoints')
+  describe('@isPlaceholderVisible', () => {
+    context('when placeholder parameter', () => {
+      it('is valid', sinon.test(function () {
 
-      instance.checkPlaceholder(instance.$element[1])
+        const spy = this.spy(instance, 'isPlaceholderVisible')
 
-      expect(instance.parseBreakpoints.calledOnce).to.be.true
-    }))
+        expect(spy.calledWith(instance.element[0])).to.be.truthy
+      }))
+
+      it('if not valid', sinon.test(function () {
+        const spy = this.spy(instance, 'isPlaceholderVisible')
+
+        instance.isPlaceholderVisible('abc123')
+
+        expect(spy.returned(false)).to.be.true
+      }))
+    })
   })
 
-  describe('parseBreakpoints', () => {
-    it('should set the [src] attr based on the window width', sinon.test(function () {
-      instance.windowWidth = 1000
+  describe('@renderImage', () => {
+    it('should return false if don\'t send placeholder parameter', sinon.test(function () {
```

Comment:
Ao meu ver, aqui cabe o uso de dois contextos.
Além disso, a descrição do contexto poderia ser algo tipo `when the placeholder parameter is not sent`.